forge-gui/res/cardsfolder/upcoming/the_beast_deathless_prince.txt

```diff
@@ -0,0 +1,13 @@
+Name:The Beast, Deathless Prince
+ManaCost:2 B R
+Types:Legendary Creature Demon
+PT:6/6
+K:ETBReplacement:Other:ETBTapped
+T:Mode$ SpellCast | ValidCard$ Card.Self | Execute$ TrigGainControl | TriggerDescription$ When you cast this spell, gain control of target creature until end of turn. Untap it. It gains menace and haste until end of turn.
+SVar:TrigGainControl:DB$ GainControl | ValidTgts$ Creature | LoseControl$ EOT | Untap$ True | AddKWs$ Haste & Menace
+SVar:ETBTapped:DB$ Tap | Defined$ Self | SubAbility$ DBAddCounter | ETB$ True | SpellDescription$ NICKNAME enters the battlefield tapped with six stun counters on it. (If it would become untapped, remove a stun counter from it instead.)
+SVar:DBAddCounter:DB$ PutCounter | Defined$ Self | ETB$ True | CounterType$ STUN | CounterNum$ 6
+T:Mode$ DamageDone | ValidSource$ Creature | ValidTarget$ Player.SourceCardOwner | CombatDamage$ True | TriggerZones$ Battlefield | Execute$ TrigUntap | TriggerDescription$ Whenever a creature deals combat damage to its owner, untap NICKNAME and draw a card.
```

Comment:
The ValidTarget$ field in the DamageDone trigger doesn't work. Right now it's triggering when any creature deals damage to any player.

Reply:
Yeah Trigger params can be a bit twisty

Try `ValidTarget$ TriggeredAttackerOwner`

Reply:
If you use discord, you might want to join: https://discord.gg/EwjzpGdc

Feel free to ping me with card scripting questions

Reply:
Tried ValidTarget$ TriggeredAttackerOwner and ValidTarget$ Player.TriggeredAttackerOwner; neither worked.

Reply:
@Northmoc
triggering objects aren't set unless test is successful

We may need to add some engine support, but you can keep this up for that

Reply:
I've tried a few other tricks but I think this will need engine support.

These WHO cards are pretty wild and many are needing code tweaks beyond just writing the card scripts.

Reply:
I think I'm really close to sorting this 🤞